torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -1400,13 +1249,30 @@ void AliasDb::rebuildWriteCache() const {
   isWriteCacheStale_ = false;
 }
 
-ValueSet AliasDb::getMemoryLocations(const Value* v) const {
+// Given a future value `v`, figure out what the "original" futures that
```

Comment:
Discussed a bit in person. It doesn't seem like this can always be accurate, since it possible to do, e.g.
```
a = [fut]
if <halting problem>:
  a.append(fut2)
b = a.pop(rand(2)).wait()
```
So futures cannot always be traced to a fork operation. Handling them correctly seems similar to better analysis of what may/may not be in a list. So maybe we should just implement both at the same time in another PR. For now, we can treat it like list extraction.